drivers/at_cmd/at_cmd.c

```diff
@@ -289,12 +384,12 @@ int at_cmd_write(const char *const cmd,
 
 void at_cmd_set_notification_handler(at_cmd_handler_t handler)
 {
+	(void)append_notif_handler(&notif_handler_list, handler);
```

Comment:
I think that we should break the old API and return an error code in case of failure. All in all this function can fail now.

Reply:
Should the modules calling this API be modified in this PR, too?
They include lte_lc.c, at_host.c and modem_info.c.

Reply:
If you feel you have time for that, then go ahead, I don't push for it though, it can be done later on.

Reply:
I would propose doing it later.